plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultQueryBuilder.java

```diff
@@ -357,10 +358,9 @@ protected String toPredicate(JdbcClient client, ConnectorSession session, Connec
             disjuncts.add(toPredicate(client, session, column, jdbcType, type, writeFunction, "=", getOnlyElement(singleValues), accumulator));
         }
         else if (singleValues.size() > 1) {
-            for (Object value : singleValues) {
-                accumulator.accept(new QueryParameter(jdbcType, type, Optional.of(value)));
-            }
-            String values = Joiner.on(",").join(nCopies(singleValues.size(), writeFunction.getBindExpression()));
+            int normalizedLength = normalizedInListLength(session, singleValues.size());
+            acceptNormalizedValues(normalizedLength, singleValues, jdbcType, type, accumulator);
+            String values = Joiner.on(",").join(nCopies(normalizedLength, writeFunction.getBindExpression()));
```

Comment:
Move the logic to a new method (like `toPredicate` above) and inline `acceptNormalizedValues`